### PR TITLE
feat(1415): Add guide for subscribed repo configuration

### LINKED ISSUE #1415: Notify SCM changes from external repositories
## Context

There are many pipelines whose builds are dependent on source changes happening across  external SCM repositories in addition to repository from which it was created.
For example `screwdriver/screwdriver-release` pipeline which needs to be built whenever a new `tag` or `release` on `screwdriver/screwdriver` or `screwdriver/ui` repositories.

## Objective

1. Add webhooks to subscribed pipelines and when webhook is triggered it should match the subscriber pipelines which are subscribed to it.
    1. If user does not have permission to add webhooks on any of the subscribed repositories, display appropriate message guiding the user with manual addition of the webhooks. 
1. SCM change in the subscribed repository should be captured in  `metadata` of the build so that subscriber pipelines can do necessary code checkouts or build commands based on change
1. It is the responsibility of subscriber pipeline to do the necessary code checkouts. [TBD]
    1.  Git Submodules can be used for this without having SD implement the logic

### Config

```

subscribe:
   scmUrls:
      - git@github.com:minz1027/test.template.git: [~commit, ~tags, ~release]
      - git@github.com:minz1027/quickstart-generic.git#master: [~commit, ~tags, ~release]
jobs:
  main:
    requires: [~subscribe, ~commit]
```

## References

1. Trigger on release/tags - https://github.com/screwdriver-cd/screwdriver/issues/823

### PR DESCRIPTION
**Pull request by supra08**
## Context

Fixes screwdriver-cd/screwdriver#1415

## Objective

This PR adds a guide for configuring subscribed repositories.

## References

screwdriver-cd/screwdriver#1415

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: docs/user-guide/configuration/workflow.md [hunk 1]
```diff
@@ -379,3 +381,31 @@ jobs:
 ```
 
 Example repo: <https://github.com/screwdriver-cd-test/workflow-detached-example>
+
+## Subscribed SCM Notifications
+You can subscribe to external repositories so that the builds are triggered in the pipeline whenever there are changes in those external repositories. The pipeline can be configured to subscribe to webhook notifications of standard events such as `~pr`, `~commit`, `~tag` and `~release`. The jobs need to be independently configured to respond to the subscribed event(s).
```
Comment (tkyi) [inline docs/user-guide/configuration/workflow.md]:
```suggestion
You can subscribe to external repositories so builds are triggered in your pipeline whenever there are changes in those external repositories. You can configure your pipeline to subscribe to webhook notifications for events such as `~pr`, `~commit`, `~tag` and `~release`. Each job needs to be independently configured to respond to the subscribed event(s).
```

File: docs/user-guide/configuration/workflow.md [hunk 2]
```diff
@@ -379,3 +381,31 @@ jobs:
...
+
+### Example
+In the following example, we can add the repositories to subsribe to in the `scmUrls` sections of the `subscribe` object. Then we need to specify the events that it is subscribing to, which here are `~commit` and `~pr`. It is to be noted that the webhooks are automatically registered to the repositories at the time of pipeline creation.
```
Comment (tkyi) [inline docs/user-guide/configuration/workflow.md]:
```suggestion
In the following example, we can add repositories to subscribe to in the `scmUrls` sections of the `subscribe` object. Then we need to specify the events to subscribing to, which are `~commit` and `~pr` in this example. Note that webhooks are automatically registered to the repositories at the time of pipeline creation.
```

For this last sentence, are you saying the user doesn't need to manually add any webhooks?

Comment (supra08) [reply docs/user-guide/configuration/workflow.md]: Yes, the webhooks are added automatically like it is being added to source repo.

File: docs/user-guide/configuration/workflow.md [hunk 3]
```diff
@@ -379,3 +381,31 @@ jobs:
 ```
 
 Example repo: <https://github.com/screwdriver-cd-test/workflow-detached-example>
+
+## Subscribed SCM Notifications
+You can subscribe to external repositories so builds are triggered in your pipeline whenever there are changes in those external repositories. You can configure your pipeline to subscribe to webhook notifications for events such as `~pr`, `~commit`, `~tag` and `~release`. Each job needs to be independently configured to respond to the subscribed event(s).
+
+### Example
+In the following example, we can add repositories to subscribe to in the `scmUrls` sections of the `subscribe` object. Then we need to specify the events to subscribing to, which are `~commit` and `~pr` in this example. Note that webhooks are automatically registered to the repositories at the time of pipeline creation.
```
Comment (tkyi) [inline docs/user-guide/configuration/workflow.md]:
```suggestion
In the following example, we can add repositories to subscribe to in the `scmUrls` sections of the `subscribe` object. Then we need to specify the events to subscribe to, which are `~commit` and `~pr` in this example. Note that webhooks are automatically registered to the repositories at the time of pipeline creation.
```

**Review by jithine — APPROVED** [review]

**Review by tkyi — APPROVED** [review]

**Pull request merged**